clang/test/SemaCXX/cheri/cheri-capability-qualifier-dependent.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %cheri_cc1 -o - %s -fsyntax-only -verify
+
+template<typename T>
+T __capability test_template(T p) {
+  // expected-note@-1 {{candidate template ignored: substitution failure [with T = int]: __capability only applies to pointers; type here is 'int'}}
+  // expected-note@-2 {{candidate template ignored: substitution failure [with T = int]: __capability only applies to pointers; type here is 'int'}}
+  T __capability c = (__cheri_tocap T __capability)p;
+  return c;
+}
+
+template void * __capability test_template<void *>(void *);
```

Comment:
```suggestion
template void *__capability test_template<void *>(void *);
```
LLVM places qualifiers immediately after the *

Reply:
It does, but that's not actually how we canonicalise `* __capability` in LLVM; in the AST dumping and Sema diagnostics we include the space (and I think it looks really weird without). Also:
```
waltham:llvm-project jrtc4% git grep ' \* __capability' clang/test | wc -l
437
waltham:llvm-project jrtc4% git grep '[^ ]\* __capability' clang/test | wc -l
109
waltham:llvm-project jrtc4% git grep '\*__capability' clang/test | wc -l 
290
```
so the current formatting is actually the most common.

Reply:
Yeah I don't really care. Feel free to use whatever you prefer. I just use whatever clang-format generates, but I do find using spaces for __capability but not for const a bit odd.

Reply:
Well, clang-format only generates that because you told it to :)